extensions/ql-vscode/src/view/common/icon/CodiconButton.tsx

```diff
@@ -0,0 +1,44 @@
+import * as React from "react";
+import { styled } from "styled-components";
+import classNames from "classnames";
+
+const StyledButton = styled.button`
+  background: none;
+  color: var(--vscode-textLink-foreground);
+  border: none;
+  cursor: pointer;
+  padding: 0;
+  vertical-align: text-bottom;
+
+  &:disabled {
+    color: var(--vscode-disabledForeground);
+    cursor: default;
+  }
+`;
+
+export const CodiconButton = ({
```

Comment:
Is `<VSCodeButton appearance="icon"><Codicon .../></VSCodeButton>` not able to do what we want here? What are the challenges that require a new button component?

I'm not against adding a new component, but I just want to verify that it is needed.

Reply:
Thanks, I didn't know that this prop existed. I'll switch to use this instead. Unfortunately it makes testing a bit harder (because they're not native components), but it does mean we'll have less custom components to maintain.